app/controllers/application_controller.rb

```diff
@@ -88,7 +88,14 @@ def set_locale
 
   def redirect_unless_admin
     unless current_user.admin?
-      redirect_to stream_url, :notice => 'you need to be an admin to do that'
+      redirect_to stream_url, notice: "you need to be an admin to do that"
+      return
+    end
+  end
+
+  def redirect_unless_admin_or_moderator
+    unless current_user.moderator? || current_user.admin?
```

Comment:
But that would drop the message below @jhass 

@zaziemo Think about what happens when you would like to add another role, say "super_admin"? Then the method name would need to change to "redirect_unless_admin_or_moderator_or_super_admin". And then you'd like to another role and all hell would break loose :)

I'd rather go with something flexible like this:

```Ruby
def authorize
  return if user_allowed?
  redirect_to stream_url, notice: "you don't have the necessary rights to do that"
end

def user_allowed?
  current_user.moderator? || current_user.admin?
end
```

Overall this is still not the best design since i'd rather go with something like "user has_many roles" or a gem that abstracts away role management but I guess thats out of the scope of this PR..;)